cmd/newallocation.go

```diff
@@ -150,16 +153,43 @@ var newallocationCmd = &cobra.Command{
 				log.Fatal("Error fetching cost: ", err)
 			}
 			log.Print("Cost for the given allocation: ", zcncore.ConvertToToken(minCost))
-		} else {
-			allocationID, err := sdk.CreateAllocation(*datashards, *parityshards,
+
+			return
+		}
+
+		var owner string
+		if flags.Changed("owner") {
+			if owner, err = flags.GetString("owner"); err != nil {
+				log.Fatal("invalid owner value: ", err)
+			}
+		}
+		var allocationID string
+		if len(owner) == 0 {
+			allocationID, err = sdk.CreateAllocation(*datashards, *parityshards,
 				*size, expireAt, readPrice, writePrice, mcct, lock)
 			if err != nil {
 				log.Fatal("Error creating allocation: ", err)
 			}
-			log.Print("Allocation created: ", allocationID)
-			storeAllocation(allocationID)
+		} else {
+			var ownerPublicKey string
+			if flags.Changed("owner") {
+				if ownerPublicKey, err = flags.GetString("owner_public_key"); err != nil {
+					log.Fatal("invalid owner public key: ", err)
+				}
+				if len(ownerPublicKey) == 0 {
+					log.Fatal("must provide owner public key, when creating an allocation for another")
+				}
+			}
+
+			allocationID, err = sdk.CreateAllocationForOwner(owner, ownerPublicKey, *datashards, *parityshards,
```

Comment:
There is tricky way to do this with sctransaction by replacing owner and ownerpublicKey in payload - chain accepts this kind of transaction creates allocation for `another` - this is how 0Box creating them now. Not sure if owner should be added during this `tricky` calls

Reply:
Not clear what change you are suggesting?